sdk/src/client/secret/ledger_nano.rs

```diff
@@ -151,13 +151,17 @@ impl SecretManage for LedgerSecretManager {
         let lock = self.mutex.lock().await;
 
         // get ledger
-        let ledger = get_ledger(coin_type, bip32_account, self.is_simulator)?;
+        let ledger = get_ledger(coin_type, bip32_account, self.is_simulator).map_err(Error::from)?;
 
         let addresses = if options.ledger_nano_prompt {
             // and generate a single address that is shown to the user
-            ledger.get_addresses(true, bip32, address_indexes.len())?
+            ledger
+                .get_addresses(true, bip32, address_indexes.len())
+                .map_err(Error::from)?
         } else {
-            ledger.get_addresses(false, bip32, address_indexes.len())?
+            ledger
+                .get_addresses(false, bip32, address_indexes.len())
+                .map_err(Error::from)?
```

Comment:
I think this can be DRYed by putting `options.ledger_nano_prompt` as the first argument of `.get_addresses(..)`?